crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -350,6 +350,16 @@ pub unsafe trait WorldQuery: for<'w> WorldQueryGats<'w> {
         change_tick: u32,
     ) -> <Self as WorldQueryGats<'w>>::Fetch;
 
+    /// This function is safe to call if `Self:ReadOnlyWorldQuery` holds.
+    ///
+    /// # Safety
+    /// While calling this method on its own cannot cause UB it is marked `unsafe` as the caller must ensure
+    /// that the returned value is not used in any way that would cause two `QueryItem<Self>` for the same
+    /// `archetype_index` or `table_row` to be alive at the same time.
+    unsafe fn clone_fetch<'w>(
```

Comment:
I feel like we probably want a safe read-only version of this too.

I'm not sure if it's immediately useful though.

Reply:
I don't think it really matters, none of the call sites seem to know that `Self: ReadOnlyWorldQuery` holds, it'd be trivial to write a safe function later if we need it so i'm not bothered about this